libs/domains/services/data-access/src/lib/utils/refacto-service-payload.ts

```diff
@@ -145,30 +143,36 @@ export function refactoDatabase(database: Database) {
 }
 
 export function refactoHelm(helm: Helm) {
-  const sourceProvider = (helm?.source as HelmResponseAllOfSourceOneOf).git ? 'GIT' : 'HELM_REPOSITORY'
-
-  const source = match(sourceProvider)
-    .with('GIT', () => {
-      const gitRepository = (helm?.source as HelmResponseAllOfSourceOneOf).git?.git_repository
-      return {
-        git_repository: {
-          url: gitRepository?.url ?? '',
-          branch: gitRepository?.branch,
-          root_path: gitRepository?.root_path,
-          git_token_id: gitRepository?.git_token_id,
-        },
+  const source = match(helm)
+    .with(
+      {
+        source: P.when(isHelmGitSource),
+      },
+      ({ source: { git } }) => {
+        return {
+          git_repository: {
+            url: git?.git_repository?.url ?? '',
```

Comment:
Do you still need nullish check on `git` and `repository` ?

Reply:
Yes needed because I can't have null url